controllers/policyserver_controller_test.go

```diff
@@ -283,4 +282,78 @@ var _ = Describe("PolicyServer controller", func() {
 		})
 	})
 
+	When("creating a PolicyServer with requests and no limits", func() {
+		policyServerName := newName("policy-server")
+		policyServer := policyServerFactory(policyServerName)
+		policyServer.Spec.Limits = corev1.ResourceList{
+			"cpu":    resource.MustParse("1Gi"),
+			"memory": resource.MustParse("500Mi"),
+		}
+
+		It("should create the PolicyServer pod with the limits and the requests", func() {
+			By("creating the PolicyServer")
+			Expect(
+				k8sClient.Create(ctx, policyServer),
+			).To(haveSucceededOrAlreadyExisted())
+
+			By("creating a deployment with limits and requests set")
+			Eventually(func(g Gomega) error {
+				deployment, err := getTestPolicyServerDeployment(policyServerName)
+				if err != nil {
+					return err
+				}
+
+				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
+				By("setting the requests to the same value as the limits")
+				Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Requests).To(Equal(policyServer.Spec.Limits))
```

Comment:
```suggestion
				g.Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Limits).To(Equal(policyServer.Spec.Limits))
				By("setting the requests to the same value as the limits")
				g.Expect(deployment.Spec.Template.Spec.Containers[0].Resources.Requests).To(Equal(policyServer.Spec.Limits))
```

Just to avoid [this](https://github.com/kubewarden/kubewarden-controller/issues/705)